core/artwork/artwork_internal_test.go

```diff
@@ -211,27 +215,86 @@ var _ = Describe("Artwork", func() {
+				Expect(err).ToNot(HaveOccurred())
+
+				img, format, err := image.Decode(r)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(format).To(Equal("png"))
+				Expect(img.Bounds().Size().X).To(Equal(15))
+				Expect(img.Bounds().Size().Y).To(Equal(15))
+			})
+			It("returns a JPEG if original image is not a PNG", func() {
+				conf.Server.CoverArtPriority = "cover.jpg"
+				r, _, err := aw.Get(context.Background(), alMultipleCovers.CoverArtID(), 200, false)
+				Expect(err).ToNot(HaveOccurred())
+
+				img, format, err := image.Decode(r)
+				Expect(format).To(Equal("jpeg"))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(img.Bounds().Size().X).To(Equal(200))
+				Expect(img.Bounds().Size().Y).To(Equal(200))
+			})
 		})
-		It("returns a JPEG if original image is not a PNG", func() {
-			conf.Server.CoverArtPriority = "cover.jpg"
-			r, _, err := aw.Get(context.Background(), alMultipleCovers.CoverArtID(), 200, false)
-			Expect(err).ToNot(HaveOccurred())
-
-			img, format, err := image.Decode(r)
-			Expect(format).To(Equal("jpeg"))
-			Expect(err).ToNot(HaveOccurred())
-			Expect(img.Bounds().Size().X).To(Equal(200))
-			Expect(img.Bounds().Size().Y).To(Equal(200))
+		When("When square is true", func() {
+			var alCover model.Album
+
+			DescribeTable("resize",
+				func(format string, landscape bool, size int) {
+					coverFileName := "cover." + format
+					dirName := createImage(format, landscape, size)
+					alCover = model.Album{
+						ID:         "444",
+						Name:       "Only external",
+						ImageFiles: filepath.Join(dirName, coverFileName),
+					}
+					ds.Album(ctx).(*tests.MockAlbumRepo).SetData(model.Albums{
+						alCover,
+					})
+
+					conf.Server.CoverArtPriority = coverFileName
+					r, _, err := aw.Get(context.Background(), alCover.CoverArtID(), size, true)
+					Expect(err).ToNot(HaveOccurred())
+
+					img, format, err := image.Decode(r)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(format).To(Equal("png"))
+					Expect(img.Bounds().Size().X).To(Equal(size))
+					Expect(img.Bounds().Size().Y).To(Equal(size))
+				},
+				Entry("portrait png image", "png", false, 200),
+				Entry("landscape png image", "png", true, 200),
+				Entry("portrait jpg image", "jpg", false, 200),
+				Entry("landscape jpg image", "jpg", true, 200),
+			)
 		})
 	})
 })
+
+func createImage(format string, landscape bool, size int) string {
+	var img image.Image
+
+	if landscape {
+		img = image.NewRGBA(image.Rect(0, 0, size, size/2))
+	} else {
+		img = image.NewRGBA(image.Rect(0, 0, size/2, size))
+	}
+
+	tmpDir := GinkgoT().TempDir()
+	switch format {
+	case "png":
+		f, _ := os.Create(filepath.Join(tmpDir, "cover.png"))
+		defer f.Close()
+		_ = png.Encode(f, img)
+	case "jpg":
+		f, _ := os.Create(filepath.Join(tmpDir, "cover.jpg"))
+		defer f.Close()
+		_ = jpeg.Encode(f, img, &jpeg.Options{Quality: 75})
+	}
+	// save variable img to variable f
```

Comment:
This comment does not seem to be in the right place, maybe it is not even needed?

Reply:
Was trying to see what copilot would do and forgot to remove it. 
Oops.